src/aspire/volume/volume_synthesis.py

```diff
@@ -0,0 +1,303 @@
+from aspire.utils.random import Random, randn
+from aspire.volume import Volume
+
+
+class SyntheticVolumeBase(abc.ABC):
+    def __init__(self, L, C, symmetry, seed=None, dtype=np.float64):
+        self.L = L
+        self.C = C
+        self.symmetry = symmetry
+        self.seed = seed
+        self.dtype = dtype
+
+    @abc.abstractmethod
+    def generate(self):
+        """
+        Called to generate and return the synthetic volumes.
+
+        Each concrete subclass should implement this.
+        """
+
+    @abc.abstractmethod
+    def _check_symmetry(self):
+        """
+        Called to check that volumes are instantiated with compatible symmetry type.
+
+        Each concrete subclass should implement this.
+        """
+
+    def __repr__(self):
+        return f"{self.__dict__}"
+
+
+class LegacyVolume(SyntheticVolumeBase):
+    """
+    Legacy ASPIRE Volume constructed of random 3D Gaussian blobs.
+
+    Suffers from too large point variances.
+    Included for migration of legacy unit tests.
+    """
+
+    def __init__(self, L, C=2, symmetry=None, K=16, seed=None, dtype=np.float64):
+        """
+        :param L: Resolution of the Volume(s) in pixels.
+        :param C: Number of Volumes to generate.
+        :param symmetry: Must be None for LegacyVolume.
+        :param K: Number of Gaussian blobs used to construct the Volume(s).
+        :param seed: Random seed for generating random Gaussian blobs.
+        :param dtype: dtype for Volume(s).
+        """
+        super().__init__(L, C, symmetry, seed=seed, dtype=dtype)
+        self.K = K
+        self._check_symmetry()
+
+    def generate(self):
+        """
+        Called to generate and return an ASPIRE LegacyVolume as a Volume instance.
+        """
+        return gaussian_blob_vols(
+            L=self.L,
+            C=self.C,
+            symmetry=self.symmetry,
+            seed=self.seed,
+            dtype=self.dtype,
+        )
+
+    def _check_symmetry(self):
+        """
+        Checks that `symmetry` is set to None.
+        """
+        if self.symmetry is not None:
+            raise ValueError("symmetry must be None for LagacyVolume")
+
+
+class SyntheticVolume(SyntheticVolumeBase):
+    """
+    Synthetic Volume constructed of random 3D Gaussian blobs with compact support in the unit sphere.
+    """
+
+    def __init__(self, L, C, symmetry, K=16, seed=None, dtype=np.float64):
+        """
+        :param L: Resolution of the Volume(s) in pixels.
+        :param C: Number of Volumes to generate.
+        :param symmetry: Volume symmetry. None or a string "Cn", n an integer.
+        :param K: Number of Gaussian blobs used to construct the Volume(s).
+        :param seed: Random seed for generating random Gaussian blobs.
+        :param dtype: dtype for Volume(s)
+        """
+        super().__init__(L, C, symmetry, seed=seed, dtype=dtype)
+        self.K = K
+        self._check_symmetry()
+
+    def generate(self):
```

Comment:
Just making a note that this is performing two steps,

generating a volume
enforcing compactness